static/app/views/releases/list/mobileBuilds.tsx

```diff
@@ -0,0 +1,143 @@
+import {useEffect, useMemo, useRef, useState} from 'react';
+
+import {Stack} from '@sentry/scraps/layout';
+
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {normalizeDateTimeParams} from 'sentry/components/organizations/pageFilters/parse';
+import {PreprodBuildsTable} from 'sentry/components/preprod/preprodBuildsTable';
+import SearchBar from 'sentry/components/searchBar';
+import {t} from 'sentry/locale';
+import type {Organization} from 'sentry/types/organization';
+import {useApiQuery, type UseApiQueryResult} from 'sentry/utils/queryClient';
+import {decodeScalar} from 'sentry/utils/queryString';
+import type RequestError from 'sentry/utils/requestError/requestError';
+import useLocationQuery from 'sentry/utils/url/useLocationQuery';
+import {useDebouncedValue} from 'sentry/utils/useDebouncedValue';
+import {useLocation} from 'sentry/utils/useLocation';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import type {ListBuildsApiResponse} from 'sentry/views/preprod/types/listBuildsTypes';
+
+type Props = {
+  organization: Organization;
+  projectSlug: string | undefined;
+};
+
+export default function MobileBuilds({organization, projectSlug}: Props) {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const {cursor, query: urlSearchQuery} = useLocationQuery({
+    fields: {
+      cursor: decodeScalar,
+      query: decodeScalar,
+    },
+  });
+  const [pendingSearchQuery, setPendingSearchQuery] = useState<string | undefined>(
+    undefined
+  );
+  const searchInputValue = pendingSearchQuery ?? urlSearchQuery ?? '';
+  const debouncedSearchInput = useDebouncedValue(searchInputValue);
+  const previousUrlSearchQuery = useRef(urlSearchQuery);
+
+  useEffect(() => {
+    if (previousUrlSearchQuery.current === urlSearchQuery) {
+      return;
+    }
+
+    previousUrlSearchQuery.current = urlSearchQuery;
+    setPendingSearchQuery(undefined);
+  }, [urlSearchQuery]);
+
+  useEffect(() => {
+    if (pendingSearchQuery === undefined) {
+      return;
+    }
+
+    if (debouncedSearchInput === (urlSearchQuery ?? '')) {
+      return;
+    }
+
+    navigate(
+      {
+        pathname: location.pathname,
+        query: {
+          ...location.query,
+          query: debouncedSearchInput.trim() || undefined,
+          cursor: undefined,
+        },
+      },
+      {replace: true}
+    );
+  }, [debouncedSearchInput, location, navigate, pendingSearchQuery, urlSearchQuery]);
```

Comment:
nuqs might be a good one to bring in here. it's new stuff we've added
https://develop.sentry.dev/frontend/url-state/#nuqs